### PR TITLE
Add snapshot testing infrastructure

### PR DESCRIPTION
**Pull request by matthew-healy**
This PR adds snapshot testing infrastructure using the `cargo-insta` crate & accompanying binary.

The README file added in https://github.com/tweag/nickel/pull/1034/commits/7251b20d1ceb9f220c49399a61b4779f1067a2a1 includes detailed instructions for interacting with the tool.

### CODE REVIEW

**Review by vkleen — APPROVED** [review]
This looks like a very good start for the snapshot testing project. We'll need more tests down the line, but this should give us the infrastructure to do it easily 👍

File: tests/snapshot/README.md [hunk 1]
```diff
@@ -0,0 +1,47 @@
+# Snapshot testing
+
+Nickel snapshot tests are run by invoking the most recently built Nickel binary
+(located via the `CARGO_BIN_EXE_nickel` environment variable) with the Nickel
```
Comment (yannham) [inline tests/snapshot/README.md:4]: Is `nickel` in lowercase on purpose, in `CARGO_BIN_EXE_nickel`? Or is it a typo?

**Review by yannham — APPROVED** [review]
LGTM. Kudos for the simple and clear documentation :+1:

File: tests/snapshot/README.md (hunk 1 above)
Comment (matthew-healy) [reply tests/snapshot/README.md:4]: It's deliberate - if you look at `CARGO_BIN_EXE_<name>` in [the docs](https://doc.rust-lang.org/cargo/reference/environment-variables.html) it says:

> The <name> is the name of the binary target, exactly as-is. For example, CARGO_BIN_EXE_my-program for a binary named my-program.

**Comment by torhovland** [comment]
Cool!

**Pull request merged**